toml-to-ical-bin/src/main.rs

```diff
@@ -102,7 +111,13 @@ impl io::Write for Output {
 fn generate() -> Result<(), Error> {
     let opts = Opt::parse();
 
-    let calendar = Calendar::load::<Session>(&opts.input)?;
+    if !opts.input.is_file() {
+        eprintln!("ERROR: Input {:?} is not a file.", opts.input);
+        std::process::exit(1);
+    }
+    let base_path = opts.input.parent().unwrap();
```

Comment:
Is it true that a file always has a parent? 🤔

Reply:
And likewise can you map this to a variant of `Error` and use `?`?

Reply:
Regarding whether a file always has a parent, my reading of the docs was (and is, upon re-review), "yes", but I'm obviously interested in any counter-examples.

The docs [say](https://doc.rust-lang.org/std/path/struct.Path.html#method.parent):

> Returns `None` if the path terminates in a root or prefix, or if it’s the empty string.

A root such as "/" cannot be a file.  Neither can the empty string.  That just leaves a [`Prefix`](https://doc.rust-lang.org/std/path/enum.Prefix.html), which is a Windows-only thing.  It doesn't seem any of those can be files either.

If we think this analysis is correct, then it would seem appropriate to `unwrap` since a crash due to this would be a violation of our expectations of how things should work.

Reply:
I've added a comment here describing why this is (believed to be) a valid use of `unwrap`.